### PR TITLE
Speed up rendering of RGB images

### PR DESCRIPTION
**Pull request by wonder-sk**
In my simple test case this made map rendering of RGB satellite image tiles
go down by ~50% from ~40ms per tile to ~20ms per tile (in debug version, ahem)

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

File: src/core/raster/qgsrasterblock.h [hunk 1]
```diff
@@ -196,6 +196,20 @@ class CORE_EXPORT QgsRasterBlock
...
     double value( qgssize index ) const;
 
+    /**
+     * Gives direct access to the raster block data.
+     * The data type of the block must be Qgis::Byte otherwise it returns null pointer.
+     * Useful for most efficient read access.
+     * \note not available in Python bindings
+     * \since QGIS 3.4
+     */
+    const quint8 *byteData() const SIP_SKIP
```
Comment (m-kuhn) [inline src/core/raster/qgsrasterblock.h]: Candidate for `inline`? Even though most compiler will happily ignore this and do what's right I think it's the correctthing to do here.

Comment (nyalldawson) [reply src/core/raster/qgsrasterblock.h]: My understanding is that inline is totally ignored by modern compilers.

**Comment by m-kuhn** [comment]
Looking great!

File: src/core/raster/qgsrasterblock.h (hunk 1 above)
Comment (m-kuhn) [reply src/core/raster/qgsrasterblock.h]: Ok, point taken. Merge ahead.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**